services/worker-manager/src/api.js

```diff
@@ -474,6 +474,67 @@ builder.declare({
   return res.reply(worker.serializable());
 });
 
+builder.declare({
+  method: 'post',
+  route: '/workers/:workerPoolId:/:workerGroup/:workerId',
+  name: 'updateWorker',
+  title: 'Update an existing Worker',
+  category: 'Workers',
+  stability: APIBuilder.stability.stable,
+  input: 'create-worker-request.yml',
+  output: 'worker-full.yml',
+  // note that this pattern relies on workerGroup and workerId not containing `/`
+  scopes: 'worker-manager:update-worker:<workerPoolId>/<workerGroup>/<workerId>',
+  cleanPayload: cleanCreatePayload,
+  description: [
+    'Update an existing worker.  The precise behavior of this method depends',
+    'on the provider implementing the given worker pool.  Some providers',
+    'do not support updating workers at all, and will return a 400 error.',
+  ].join('\n'),
```

Comment:
Can the description provide a bit of context why this was added, and the different types of implementations one might expect? Or maybe link to provider docs for more information. The current wording leaves a lot of open questions.

Reply:
This matches the createWorker call.  I can provide a pointer to the documentation for each provider type.

Reply:
I agree. Seeing examples of scenarios where one would call this method will be helpful in general.